src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/AssemblyBuilderImpl.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+using System.Reflection.PortableExecutable;
+
+namespace System.Reflection.Emit
+{
+    internal sealed class AssemblyBuilderImpl : AssemblyBuilder
+    {
+        private bool _previouslySaved;
+        private readonly AssemblyName _assemblyName;
+        private readonly Assembly _coreAssembly;
+        private ModuleBuilderImpl? _module;
+
+        internal AssemblyBuilderImpl(AssemblyName name, Assembly coreAssembly, IEnumerable<CustomAttributeBuilder>? assemblyAttributes)
+        {
+            ArgumentNullException.ThrowIfNull(name);
+
+            name = (AssemblyName)name.Clone();
+            _coreAssembly = coreAssembly;
+
+            if (string.IsNullOrEmpty(name.Name))
+            {
+                throw new ArgumentException(SR.Argument_NullOrEmptyAssemblyName);
+            }
+
+            _assemblyName = name;
```

Comment:
```suggestion
 
            if (string.IsNullOrEmpty(name.Name))
            {
                throw new ArgumentException(SR.Argument_NullOrEmptyAssemblyName);
            }

            _assemblyName = name;

            _coreAssembly = coreAssembly;
```
More logical ordering.